### ISSUE TITLE
Inconsistent error: "invalid parameter install_options"

### ISSUE THREAD

**Issue by ntent-ashton** [elastic/puppet-elasticsearch#610]
`Could not retrieve catalog from remote server: Error 400 on SERVER: Invalid parameter install_options on Elasticsearch_plugin[lmenezes/elasticsearch-kopf] at /etc/puppet/environments/development/modules/elasticsearch/manifests/plugin.pp:142 on node es-02-dev`

I really have no idea what is even causing this, or why it's only happening 1-4 times a day.  Line 142 in plugins.pp doesn't seem to actually be the culprit.

**Comment by tylerjl** [comment]
Hey @ntent-ashton, I haven't run across this before, and the only reference to it in Puppet docs is in [reference to development and pluginsync](https://docs.puppet.com/guides/troubleshooting.html#err-could-not-retrieve-catalog-invalid-parameter-foo-for-type-bar).

In order to reproduce I'd need your distro (and distro version), puppet version, module version, whether you're on standalone puppet or agent/master, and maybe an example puppet manifest that causes this.

**Comment by ntent-ashton** [comment]
Sorry, bad form on my part.

OS: CentOS 6.7
Puppet: 3.8.7, open-source
Module: 0.10.2
Elasticsearch: 2.3.1

We have an agent-master setup, with the additions of Hiera and Foreman.
The manifest itself is sizeable, but this is the code that installs the offending module:

``` puppet
elasticsearch::plugin{ 'lmenezes/elasticsearch-kopf':
    instances  => $profiles::elasticsearch::common::instances,
}
```

There are other modules installed as well, but for some odd reason only elasticsearch-kopf is causing the issue.  And again, it only happens 3-4 times a day, and only in my development environment.  I have a much larger cluster in a different environment that is *not* having this issue.  Go figure.

**Comment by ntent-ashton** [comment]
I searched for install_options in my entire puppet repo, only the apt, mysql, chocolatey, reboot, docker, types, and nrpe modules contain the string.

**Comment by tylerjl** [comment]
I stood up a test environment with the specs you noted (aside from puppet 3.8.7, I used 3.8.6 as I don't think 3.8.7 exists yet :smile: ) but couldn't get the behavior to exhibit (tried several times after various combinations of server stopping/starting, moving manifests/modules around, and so on.)

As it stands I can't find a way to get it to reproduce, but if you can find some steps to get the behavior to show up, I'm definitely willing to try them.

**Comment by ntent-ashton** [comment]
Haha sorry, typo. I can't reproduce it myself, and I'm completely baffled as to why it's happening in the first place. I guess mark this closed for now and I'll just try redeploying the environment.

Sent from my Android phone using Symantec TouchDown (www.symantec.com)

-----Original Message-----

**Comment by awegmann** [comment]
Hat the same issue here after updating from 0.10.1 to 0.11.0. (puppet 3.8.7, CentOS 6.7 on master, RedHat 6.7 on agent). Because I had the intuition, that this may be a caching problem I restarted the master (complete reboot of machine). 
After reboot the problem disappears.
I know, this is not an explanation for the root cause, but maybe a hint for other people having the same problem. It looks like the master didn't recognize, that the type "elasticsearch_plugin" was changed. "install_option" was added from 0.10.1 to 0.11.0 (and maybe already on 0.10.2).
